### PR TITLE
chore: make sure to run webview-preload init when running e2e tests

### PR DESCRIPTION
**Pull request by odockal**
### What does this PR do?
Let's webview-preload init method to be called when running e2e playwright tests
### What issues does this PR fix or reference?
#6740

### How to test this PR?

- [ ] Tests are covering the bug fix or the new feature

### CODE REVIEW

**Review by axel7083 — APPROVED** [review]
LGTM

**Review by benoitf — CHANGES_REQUESTED** [review]
the VITEST was for unit test but we should not include another variable for PLAYWRIGHT

**Comment by axel7083** [comment]
> the VITEST was for unit test but we should not include another variable for PLAYWRIGHT

Since the `VITEST` is defined when running with playwright, the `init` is not call when running e2e tests[^1]. I understand the need of the if condition to allow proper unit testing, however we would need some ways to have the init method called... 

IMO we would either need an ENV stating it is a unit tests or a smoke e2e to be able to achieve the expected behaviour

[^1]: https://github.com/containers/podman-desktop/issues/6740#issuecomment-2066936618

**Review by cbr7 — APPROVED** [review]

**Comment by odockal** [comment]
> the VITEST was for unit test but we should not include another variable for PLAYWRIGHT

I guess the VITEST env. var is there by default when using vitest test runner. It does not change the fact the e2e tests are different from unit ones. I welcome any other way how to make sure we do run e2e with webview-preload being correctly loaded.

other options atm: unset VITEST, set VITEST to 'e2e' and use this value to be checked in `webview-preload/index.ts`. But I am not sure about consequences of adjusting VITEST env. var.

**Review by xbabalov — APPROVED** [review]

File: packages/preload-webview/src/index.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ export const init = (): void => {
   webviewPreload.init().catch((error: unknown) => console.error('Error while initializing the exposure', error));
 };
 
-// do not call init methd in case of testing
-if (!process.env.VITEST) {
+// do not call init methd in case of testing, but call when running E2E tests, see #6740
+if (!process.env.VITEST || process.env.PLAYWRIGHT_E2E) {
```
Comment (benoitf) [inline packages/preload-webview/src/index.ts]:
```suggestion
// do not call init method in case of testing
if (import.meta.env.MODE !== 'test') {
```

File: tests/playwright/src/runner/podman-desktop-runner.ts [hunk 2]
```diff
@@ -247,9 +247,12 @@ export class PodmanDesktopRunner {
   private setupPodmanDesktopCustomFolder(): object {
     const env: { [key: string]: string } = process.env as { [key: string]: string };
     const dir = join(this._customFolder);
-    console.log(`podman desktop custom config will be written to: ${dir}`);
+    console.log(`Podman Desktop custom config will be written to: ${dir}`);
     env.PODMAN_DESKTOP_HOME_DIR = dir;
 
+    console.log('Setting PLAYWRIGHT_E2E env. variable - used in dev. mode');
+    env.PLAYWRIGHT_E2E = 'true';
+
```
Comment (benoitf) [inline tests/playwright/src/runner/podman-desktop-runner.ts]:
```suggestion
```

File: packages/preload-webview/src/index.ts (hunk 1 above)
Comment (benoitf) [reply packages/preload-webview/src/index.ts]: use import MODE instead of VITEST

as for e2e we replace the mode when compiling to not be test

**Comment by benoitf** [comment]
we're already avoiding to use test mode when building the files for e2e
https://github.com/containers/podman-desktop/blob/main/package.json#L50

so relies on TEST mode is working. (as variable is resolved at build time)

**Comment by odockal** [comment]
> we're already avoiding to use test mode when building the files for e2e https://github.com/containers/podman-desktop/blob/main/package.json#L50
> 
> so relies on TEST mode is working. (as variable is resolved at build time)

Sounds good!

**Review by benoitf — APPROVED** [review]

**Comment by odockal** [comment]
@benoitf Works nice. Can we merge?

File: tests/playwright/src/runner/podman-desktop-runner.ts [hunk 3]
```diff
@@ -247,7 +247,7 @@ export class PodmanDesktopRunner {
   private setupPodmanDesktopCustomFolder(): object {
     const env: { [key: string]: string } = process.env as { [key: string]: string };
     const dir = join(this._customFolder);
-    console.log(`podman desktop custom config will be written to: ${dir}`);
+    console.log(`Podman Desktop custom config will be written to: ${dir}`);
```
Comment (benoitf) [inline tests/playwright/src/runner/podman-desktop-runner.ts]: the two changes in this file are unrelated to this PR so I would make it part of another commit/PR

Comment (odockal) [reply tests/playwright/src/runner/podman-desktop-runner.ts]: ok.

**Review by benoitf — CHANGES_REQUESTED** [review]
it looks the changes have been omitted

**Review by benoitf — APPROVED** [review]

**Comment by odockal** [comment]
@benoitf Now it should be fine.

**Review by ScrewTSW — COMMENTED** [review]
Verified latest changes, AI Lab extension is displayed correctly

LGTM

**Review by ScrewTSW — COMMENTED** [review]
Verified latest changes, AI Lab extension is displayed correctly

LGTM
